lib/nx_inflate.c

```diff
@@ -752,6 +756,14 @@ int nx_inflate(z_streamp strm, int flush)
 	copy_stream_in(s->zstrm, s);
 	copy_stream_out(s->zstrm, s);
 
+	/* Following zlib behaviour. If there is no progress Z_BUF_ERROR
+	 * is returned. */
+	in -= s->avail_in;
+	out -= s->avail_out;
+	if (in == 0 && out == 0) return Z_BUF_ERROR;
```

Comment:
zlib also checks if `rc == Z_OK` in this case, which I think is necessary. There are many cases in the switch/case above that return `rc != Z_OK`, with the current code all of them would be replaced by `Z_BUF_ERROR` as the final return value.

Reply:
Agreed.